File: Going Dutch/MCExchangeRate.m
//
//  MCExchangeRate.m
//  We all pay
//
//  Created by Mark Cornelisse on 01/08/14.
//  Copyright (c) 2014 Mark Cornelisse. All rights reserved.
//

#import "MCExchangeRate.h"
#import "MCCurrency.h"
#import "MCPayment.h"


@implementation MCExchangeRate

#pragma mark - NSManagedObject

- (void)awakeFromInsert {
    [super awakeFromInsert];
    [self setPrimitiveValue:NSUUID.UUID.UUIDString forKey:@"uniqueID"];
    NSDate *now = NSDate.date;
    [self setPrimitiveValue:now forKey:@"dateCreated"];
    [self setPrimitiveValue:now forKey:@"dateModified"];
}

#pragma mark - NSObject

@end

File: Going Dutch/MCExchangeRate.swift
//
//  MCExchangeRate.swift
//  We all pay
//
//  Created by Mark Cornelisse on 26/08/2025.
//  Copyright © 2025 Mark Cornelisse. All rights reserved.
//

import UIKit

@objc(MCExchangeRate) public final class MCExchangeRate: NSManagedObject {
    
    // MARK: NSManagedObject
    
    public override func awakeFromInsert() {
        super.awakeFromInsert()
        
        self.setPrimitiveValue(UUID().uuidString, forKey: #keyPath(MCExchangeRate.uniqueID))
        let now = Date()
        self.setPrimitiveValue(now, forKey: #keyPath(MCEmailAddress.dateCreated))
        self.setPrimitiveValue(now, forKey: #keyPath(MCEmailAddress.dateModified))
    }
    
    // MARK: NSObject

}
